### PR TITLE
health-insurance - refactor and added new tag

### PR DESCRIPTION
**Pull request by qdd1**
# \<Description\>

Attach a link to issue if relevant

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/api/domains/health-insurance/src/lib/soap/healthInsuranceAPI.ts [hunk 1]
```diff
@@ -165,6 +164,8 @@ export class HealthInsuranceAPI {
         fyrrautgafulandkodi: inputObj.previousCountryCode,
         fyrriutgafustofnunlands: inputObj.previousIssuingInstitution ?? '',
         tryggdurfyrralandi: inputObj.isHealthInsuredInPreviousCountry,
+        tryggingaretturfyrralandi:
+          inputObj.hasHealthInsuranceRightInPreviousCountry,
```
Comment (eirikurn) [inline libs/api/domains/health-insurance/src/lib/soap/healthInsuranceAPI.ts:168]: Shouldn't this module be in @island.is/health-insurance?

File: libs/health-insurance/src/lib/bucket/bucket.service.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  }
+
+  /* */
+  async getFile(filename: string): Promise<S3.GetObjectOutput> {
+    this.logger.info('get bucket file:' + filename)
+    return new Promise((resolve, reject) => {
+      s3.createBucket(
+        {
+          Bucket: BUCKET_NAME,
+        },
```
Comment (eirikurn) [inline libs/health-insurance/src/lib/bucket/bucket.service.ts:48]: Why createBucket?

File: libs/health-insurance/src/lib/bucket/bucket.service.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        function () {
+          const params = {
+            Bucket: BUCKET_NAME,
+            Key: filename,
+          }
```
Comment (eirikurn) [inline libs/health-insurance/src/lib/bucket/bucket.service.ts:53]: You can parse these from the url using [amazon-s3-uri](https://www.npmjs.com/package/amazon-s3-uri).

File: libs/health-insurance/src/lib/bucket/bucket.service.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+import * as AWS from 'aws-sdk'
+import * as S3 from 'aws-sdk/clients/s3'
+import { Logger, LOGGER_PROVIDER } from '@island.is/logging'
+
+// TODO: get bucket name from url - STOP HARD CODE
+const BUCKET_NAME = 'island-is-dev-storage-application-system'
+const REGION = 'eu-west-1'
+
+const s3 = new AWS.S3({
+  region: REGION,
```
Comment (eirikurn) [inline libs/health-insurance/src/lib/bucket/bucket.service.ts:11]: Region should be configured from the environment/profile.

File: libs/application/template-api-modules/src/lib/modules/templates/health-insurance/health-insurance.utils.ts [hunk 5]
```diff
@@ -119,6 +119,11 @@ export const transformApplicationToHealthInsuranceDTO = (
     previousIssuingInstitution:
       extractAnswer(application.answers, 'formerInsurance.institution') ?? '',
     isHealthInsuredInPreviousCountry:
+      extractAnswer(application.answers, 'formerInsurance.registration') ==
```
Comment (dd-sdang) [inline libs/application/template-api-modules/src/lib/modules/templates/health-insurance/health-insurance.utils.ts:122]: is there a reason why the check is `==` and not `===`? :)

Comment (dd-sdang) [reply libs/application/template-api-modules/src/lib/modules/templates/health-insurance/health-insurance.utils.ts:122]: I also don't like my own naming in the frontend of this field (`formerInsurance.registration`). I will change it in a separate PR and make sure I update in this file as well.

File: libs/application/template-api-modules/src/lib/modules/templates/health-insurance/health-insurance.utils.ts [hunk 6]
```diff
@@ -119,6 +119,11 @@ export const transformApplicationToHealthInsuranceDTO = (
...
+      'yes'
+        ? 1
+        : 0,
+    hasHealthInsuranceRightInPreviousCountry:
       extractAnswer(application.answers, 'formerInsurance.entitlement') == 'yes'
```
Comment (dd-sdang) [inline libs/application/template-api-modules/src/lib/modules/templates/health-insurance/health-insurance.utils.ts:127]: Same as above. Will update the field in frontend and make sure it changes here as well.

File: libs/health-insurance/src/lib/soap/healthInsuranceAPI.ts [hunk 7]
```diff
@@ -101,7 +106,10 @@ export class HealthInsuranceAPI {
         const filename = arrAttachments[i]
```
Comment (eirikurn) [inline libs/health-insurance/src/lib/soap/healthInsuranceAPI.ts:106]: Should `attachmentsFileNames` be absolute urls to the files on AWS? You can get the URL and the original filename from `application.attachments`.

File: libs/api/domains/health-insurance/src/lib/bucket.service.ts [hunk 8]
```diff
@@ -3,7 +3,7 @@ import * as AWS from 'aws-sdk'
 import * as S3 from 'aws-sdk/clients/s3'
 import { Logger, LOGGER_PROVIDER } from '@island.is/logging'
 
-const BUCKET_NAME = 'dev-island-is-sjukra'
+const BUCKET_NAME = 'island-is-dev-storage-application-system'
```
[the remark this answers is not in the document]
Comment (qdd1) [reply libs/api/domains/health-insurance/src/lib/bucket.service.ts:6]: This doesn't need in domains but I keep it there so I could test codes without front-end.
Will be clean-up

**Review by dd-sdang — APPROVED** [review]
From my side it looks good. Just added a few comments for myself to change in frontend in a separate PR where I will update the corresponding reference here as well.

File: libs/api/domains/health-insurance/src/lib/soap/healthInsuranceAPI.ts (hunk 1 above)
Comment (qdd1) [reply libs/api/domains/health-insurance/src/lib/soap/healthInsuranceAPI.ts:168]: This module is in @island.is/health-insurance. Same as above. Will be clean up

File: libs/health-insurance/src/lib/bucket/bucket.service.ts (hunk 2 above)
Comment (qdd1) [reply libs/health-insurance/src/lib/bucket/bucket.service.ts:48]: Will be changed. I tried to get new API request changes go live so front-end could continue.
Bucket codes are not ready and will  be refactor

File: libs/health-insurance/src/lib/bucket/bucket.service.ts (hunk 3 above)
Comment (qdd1) [reply libs/health-insurance/src/lib/bucket/bucket.service.ts:53]: Thanks. Will be refactor

File: libs/health-insurance/src/lib/bucket/bucket.service.ts (hunk 4 above)
Comment (qdd1) [reply libs/health-insurance/src/lib/bucket/bucket.service.ts:11]: Same as above

File: libs/health-insurance/src/lib/soap/healthInsuranceAPI.ts (hunk 7 above)
Comment (qdd1) [reply libs/health-insurance/src/lib/soap/healthInsuranceAPI.ts:106]: I'm using original filename now.
Will have to get url and use it later when refactor Bucket codes

Comment (eirikurn) [reply libs/health-insurance/src/lib/soap/healthInsuranceAPI.ts:106]: We actually rename the file before storing it on S3 (adding a GUID prefix to deal with duplicate filenames). The real original file name is under `application.attachments `. But maybe you prefer a unique filename with guid?

**Review by eirikurn — APPROVED** [review]
Please take ownership of your template-api module in `.github/CODEOWNERS`. You can still reach out if you want code review for specific things.

**Pull request merged**